torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -119,38 +119,35 @@ using namespace torch::distributed::autograd;
 c10::intrusive_ptr<JitFuture> wrapFutureMessageInJitFuture(
     const std::shared_ptr<FutureMessage>& futureResponseMessage,
     bool hasValue) {
-  // Save and pass thread local state into the callback. The thread local state
-  // is a lightweight structure when the profiler is disabled.
-  at::ThreadLocalState tls_state;
   if (hasValue) {
     c10::intrusive_ptr<JitFuture> jitFuture =
         c10::make_intrusive<JitFuture>(PyObjectType::get());
+    std::weak_ptr wp = futureResponseMessage;
```

Comment:
I think you need the templated `weak_ptr` type (e.g. `std::weak_ptr<FutureMessage>`) to appease the compiler